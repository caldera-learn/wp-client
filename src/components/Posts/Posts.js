import React from 'react';
import PropTypes from 'prop-types';
import {Post} from "../Post/Post";
export const Posts = (props) =>  {
	return(
		<div
			className={props.className}
		>
			{props.posts.map( post => {
				return <Post
					post={post}
				/>
			})};
		</div>
	);
};

Posts.propTypes = {
	className: PropTypes.string.isRequired,
	posts: PropTypes.arrayOf(
			PropTypes.shape({
			title: PropTypes.shape({
				rendered: PropTypes.string,
			}).isRequired,
			content: PropTypes.shape({
				rendered: PropTypes.string,
			}).isRequired,
		}).isRequired
	)
};

Posts.defaultProps = {
	className: 'post-list-wrapper'
};