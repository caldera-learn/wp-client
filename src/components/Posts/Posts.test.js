import React from 'react';
import {Posts} from "./Posts";
import renderer from 'react-test-renderer';

describe('Posts component', () => {
	/** Mock WordPress post to test with**/
	const posts = [
		{
			ID: '1',
			title: {
				rendered: 'Hi Roy'
			},
			content: {
				rendered: 'Lorem ipsum, etc.'
			}
		},
		{
			ID: '2',
			title: {
				rendered: 'Hi Mike'
			},
			content: {
				rendered: 'Lorem ipsum, etc.'
			}
		},
	];


	it('Matches snapshot with basic props', () => {
		const component = renderer.create(
			<Posts
				posts={posts}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Applies className to outermost element', () => {
		const component = renderer.create(
			<Posts
				posts={posts}
				className={'food'}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

});