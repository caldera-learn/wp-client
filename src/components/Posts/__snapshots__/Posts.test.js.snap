// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts component Applies className to outermost element 1`] = `
<div
  className="food"
>
  <div
    className="post-wrapper"
  >
    <h3>
      Hi Roy
    </h3>
    <div>
      Lorem ipsum, etc.
    </div>
  </div>
  <div
    className="post-wrapper"
  >
    <h3>
      Hi Mike
    </h3>
    <div>
      Lorem ipsum, etc.
    </div>
  </div>
  ;
</div>
`;

exports[`Posts component Matches snapshot with basic props 1`] = `
<div
  className="post-list-wrapper"
>
  <div
    className="post-wrapper"
  >
    <h3>
      Hi Roy
    </h3>
    <div>
      Lorem ipsum, etc.
    </div>
  </div>
  <div
    className="post-wrapper"
  >
    <h3>
      Hi Mike
    </h3>
    <div>
      Lorem ipsum, etc.
    </div>
  </div>
  ;
</div>
`;
